etc/docker/Dockerfile

```diff
@@ -36,13 +14,23 @@ ENV RUSTUP_HOME=/opt/rust/rustup \
 
 RUN apt update && apt dist-upgrade -y
 
-RUN apt update
-
-RUN apt-get install -y --no-install-recommends git wget flex bison gperf python3 python3-pip python3-virtualenv cmake ninja-build ccache libffi-dev libssl-dev libudev-dev dfu-util libusb-1.0-0 openssh-client software-properties-common python3-venv sudo curl && \
+RUN apt-get install -y --no-install-recommends make build-essential pkg-config git wget flex bison gperf python3 python3-pip python3-virtualenv cmake ninja-build ccache libffi-dev libssl-dev libudev-dev dfu-util libusb-1.0-0 openssh-client software-properties-common python3-venv sudo curl libpixman-1-0 libslirp0 libsdl2-2.0-0 xz-utils && \
     apt-get install -y --no-install-recommends cmake && \
     apt-get install -y --no-install-recommends jq && \
     update-alternatives --install /usr/bin/python python /usr/bin/python3 10
 
+RUN set -eux; \
+    dpkgArch="$(dpkg --print-architecture)"; \
+    qemuSha="https://github.com/espressif/qemu/releases/download/esp-develop-9.0.0-20240606/qemu-esp_develop_9.0.0_20240606-checksum.sha256"; \
```

Comment:
sanity check on your heiroglyphics, so you're initializing `qemuSha` then updating it depending on the `dpkgArch`?

you're then downloading the binary, extracting the tar to  `/usr` then removing the original file?

Reply:
yes was wrong :P 

yes indeed